api/pkg/transformer/feast/bigtablestore/encoding.go

```diff
@@ -0,0 +1,462 @@
+package bigtablestore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/bigtable"
+
+	feast "github.com/feast-dev/feast/sdk/go"
+	"github.com/feast-dev/feast/sdk/go/protos/feast/serving"
+	"github.com/feast-dev/feast/sdk/go/protos/feast/types"
+	"github.com/gojek/merlin/pkg/transformer/spec"
+	"github.com/gojek/merlin/pkg/transformer/types/converter"
+	"github.com/golang/protobuf/proto"
+	"github.com/linkedin/goavro/v2"
+)
+
+// Encoder is used to encode OnlineFeaturesRequest into RowQuery
+type Encoder struct {
+	registry        CodecRegistry
+	featureSpecs    map[featureTableKey]*spec.FeatureTable
+	featureMetadata map[featureTableKey]*spec.FeatureTableMetadata
+	featureTypes    map[featureNameKey]string
+}
+
+type featureTableKey struct {
+	project string
+	table   string
+}
+
+type featureNameKey struct {
+	project string
+	feature string
+}
+
+func entityKeysToBigTable(project string, entityKeys []*spec.Entity) string {
+	keyNames := make([]string, len(entityKeys))
+	for i, e := range entityKeys {
+		keyNames[i] = e.Name
+	}
+	return project + "__" + strings.Join(keyNames, "__")
+}
+
+func feastValueToStringRepr(val *types.Value) (string, error) {
+	switch v := val.Val.(type) {
```

Comment:
Why is there no float / double value type?

Reply:
nice catch, I'll add those

Reply:
after discuss with @khorshuheng , it doesn't make sense to have double/float for entity. Also current implementation of feast serving also like this